mParticle-Apple-SDK/MPBackendController.m

```diff
@@ -1815,37 +1525,34 @@ - (void)startWithKey:(NSString *)apiKey secret:(NSString *)secret firstRun:(BOOL
         date = [NSDate date];
     }
     
-    __weak MPBackendController *weakSelf = self;
-    dispatch_async(messageQueue, ^{
+    dispatch_async([MParticle messageQueue], ^{
         [MParticle sharedInstance].persistenceController = [[MPPersistenceController alloc] init];
 
         if (shouldBeginSession) {
             [self beginSessionWithIsManual:!MParticle.sharedInstance.automaticSessionTracking date:date];
         }
         
         MPMessageBuilder *messageBuilder = [MPMessageBuilder newBuilderWithMessageType:MPMessageTypeFirstRun session:self.session messageInfo:nil];
+                
+        [self processOpenSessionsEndingCurrent:NO completionHandler:^(void) {}];
+        [self waitForKitsAndUploadWithCompletionHandler:nil];
```

Comment:
is this needed? since we'll call it again within beginUploadTimer?

Reply:
@samdozor I actually explicitly added this in based on notes from one of our conversations. :)

I believe the thinking was since the session is ending a batch is mandatory anyway, so it was better to just upload it immediately than to wait since it wouldn't matter for the "events per batch" metric anyway. But you're correct it's not strictly necessary as it will be uploaded on the next timeout (if the app doesn't crash or and isn't force killed by the user).